docs/aot.md

```diff
@@ -0,0 +1,232 @@
+# Ahead-of-time lowering and compilation
+
+JAX offers several transformations, such as `jax.jit` and `jax.pmap`, returning
+a function that is compiled and runs on accelerators or the CPU. As the JIT
+acronym indicates, all compilation happens _just-in-time_ for execution.
+
+Some situations call for _ahead-of-time_ (AOT) compilation instead. When you
+want to fully compile prior to execution time, or you want control over when
+different parts of the compilation process take place, JAX has some options for
+you.
+
+First, let's review the stages of compilation. When a function `f` output by
+{func}`jax.jit` is invoked with arguments, say `f(x, y)` where `x` and `y` are
+arrays, JAX does the following in order:
+
+1. **Specialize** `f` to properties of the arguments `x` and `y` (usually their
```

Comment:
```suggestion
1. **Stage out** a specialized version of the Python callable `f` to a jaxpr, specializing to an input type inferred from properties of the arguments `x` and `y` (usually their
```

I like the 'specialize' language so we should keep that word around, but I think 'staging out' is a more complete description and also better matches existing docs/comments throughout JAX. I say it's more complete because "specialize" by itself could mean just using `functools.partial`.